src/classes/SftpSessionHandler.ts

```diff
@@ -1,14 +1,67 @@
+import path from 'path';
+import { v4 as uuidv4 } from 'uuid';
+import ssh2 from 'ssh2';
 import { logger } from '../logger';
+import {
+  generateFileEntry,
+  getFileType,
+  generateDefaultAttributes,
+  loadPath,
+} from '../utils';
+import type {
+  Attributes,
+  FileEntry,
+  SFTPWrapper,
+} from 'ssh2';
+
+const SFTP_STATUS_CODE = ssh2.utils.sftp.STATUS_CODE;
+
+const generateHandle = (): string => uuidv4();
+
+export class SftpSessionHandler {
+  private readonly authToken: string;
+
+  private readonly sftpConnection: SFTPWrapper;
+
+  private readonly openDirectories: Map<string, FileEntry[]> = new Map();
+
+  private readonly openFiles: Map<string, Buffer> = new Map();
+
+  public constructor(
+    sftpConnection: SFTPWrapper,
+    authToken: string,
+  ) {
+    this.sftpConnection = sftpConnection;
+    this.authToken = authToken;
+  }
 
-export class SftpStreamHandler {
   /**
    * See: SFTP events (OPEN)
    * https://github.com/mscdex/ssh2/blob/master/SFTP.md
    * Also: Opening, Creating, and Closing Files
    * https://datatracker.ietf.org/doc/html/draft-ietf-secsh-filexfer-02#section-6.3
    */
-  public openHandler = (): void => {
+  public openHandler = (
+    reqId: number,
+    filename: string,
+    flags: number,
+    attrs: Attributes,
+  ): void => {
     logger.verbose('SFTP file open request (SSH_FXP_OPEN)');
+    logger.debug('Request:', {
+      reqId,
+      filename,
+      flags,
+      attrs,
+    });
+    const handle = generateHandle();
+    logger.debug(`Opening ${filename}: ${handle}`);
+    this.openFiles.set(handle, Buffer.from('content goes here'));
```

Comment:
Probably! 😂   It for sure won't last in the long run.

Honestly I think that I'm gonna merge this -- this is a weird repository since there's so much R&D and demo associated with it.